pkg/ddc/juicefs/type.go

```diff
@@ -45,19 +45,24 @@ type JuiceFS struct {
 }
 
 type Configs struct {
-	Name               string `json:"name"`
-	AccessKeySecret    string `json:"accesskeySecret,omitempty"`
-	AccessKeySecretKey string `json:"accesskeySecretKey,omitempty"`
-	SecretKeySecret    string `json:"secretkeySecret,omitempty"`
-	SecretKeySecretKey string `json:"secretkeySecretKey,omitempty"`
-	Bucket             string `json:"bucket,omitempty"`
-	MetaUrlSecret      string `json:"metaurlSecret,omitempty"`
-	MetaUrlSecretKey   string `json:"metaurlSecretKey,omitempty"`
-	TokenSecret        string `json:"tokenSecret,omitempty"`
-	TokenSecretKey     string `json:"tokenSecretKey,omitempty"`
-	Storage            string `json:"storage,omitempty"`
-	FormatCmd          string `json:"formatCmd,omitempty"`
-	QuotaCmd           string `json:"quotaCmd,omitempty"`
+	Name                string `json:"name"`
+	AccessKeySecret     string `json:"accesskeySecret,omitempty"`
```

Comment:
How about using the following configuration to make it extensible?
```golang
type Configs struct {
    Name      string   `json:"name"`
    Buckets   []Bucket `json:"buckets"`
    // other fields...
}

type Bucket struct {
    Name          string `json:"name"`
    AccessKey     string `json:"accessKey"`
    SecretKey     string `json:"secretKey"`
    // other fields...
}
```

Reply:
It's an internal config used by runtime. `configs` is all the options used by `juicefs format` or `juicefs auth`,  its quantity is almost fixed and new additions are rare. So, I think it is meaningless to make it extensible.

Reply:
How about making the dataset api  in the following way extensible and readable? like `bucketname.access-key`?

```yaml
kind: Dataset
metadata:
  name: jfsdemo
spec:
  mounts:
    - name: minio
      mountPoint: "juicefs:///demo"
      options:
        bucket: "bucket1,bucket2"
        storage: "minio"
      encryptOptions:
        - name: metaurl
          valueFrom:
            secretKeyRef:
              name: jfs-secret
              key: metaurl
        - name: bucket1.access-key
          valueFrom:
            secretKeyRef:
              name: jfs-secret
              key: access-key
        - name: bucket1.secret-key
          valueFrom:
            secretKeyRef:
              name: jfs-secret
              key: secret-key
        - name: bucket2.access-key
          valueFrom:
            secretKeyRef:
              name: jfs-secret
              key: access-key
        - name: bucket2.secret-key
          valueFrom:
            secretKeyRef:
              name: jfs-secret
              key: secret-key
```

Reply:
I think we don't support this way by default. This is only for  `bucket: "bucket1,bucket2"`

Reply:
If there is only one bucket:

```yaml
apiVersion: data.fluid.io/v1alpha1
kind: Dataset
metadata:
  name: jfsdemo
spec:
  mounts:
    - name: minio
      mountPoint: "juicefs:///"
      options:
        bucket: "http://minio:9000/minio/test"
        storage: "minio"
      encryptOptions:
        - name: metaurl
          valueFrom:
            secretKeyRef:
              name: jfs-secret
              key: metaurl
        - name: access-key
          valueFrom:
            secretKeyRef:
              name: jfs-secret
              key: access-key
        - name: secret-key
          valueFrom:
            secretKeyRef:
              name: jfs-secret
              key: secret-key
```

If there are more than one bucket:

```yaml
kind: Dataset
metadata:
  name: jfsdemo
spec:
  mounts:
    - name: minio
      mountPoint: "juicefs:///demo"
      options:
        bucket: "bucket1,bucket2"
        storage: "minio"
      encryptOptions:
        - name: metaurl
          valueFrom:
            secretKeyRef:
              name: jfs-secret
              key: metaurl
        - name: bucket1.access-key
          valueFrom:
            secretKeyRef:
              name: jfs-secret
              key: access-key
        - name: bucket1.secret-key
          valueFrom:
            secretKeyRef:
              name: jfs-secret
              key: secret-key
        - name: bucket2.access-key
          valueFrom:
            secretKeyRef:
              name: jfs-secret
              key: access-key
        - name: bucket2.secret-key
          valueFrom:
            secretKeyRef:
              name: jfs-secret
              key: secret-key
```